File: C_Sharp_code/Example/WindowsFormsApp/VB_demo21/Animal.vb
Public Class Animal
    Private _name As String

    Sub New()
        Me.New("no name")
    End Sub
    Sub New(ByVal argName As String)
        Me._name = argName
    End Sub

    Public Property Name As String
        Get
            Return _name
        End Get
        Set(ByVal value As String)
            _name = value
        End Set
    End Property
    Public Overridable Sub Talk()
        Console.WriteLine("Animal Talk")
    End Sub
End Class

File: C_Sharp_code/Example/WindowsFormsApp/CS_demo20/Animal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CS_demo20
{
    public class Animal
    {
        private string _name;

        public Animal():this("no name")
        {
            
        }
        public Animal(string argName)
        {
            this._name = argName;
        }

        public string Name
        {
            get {
                return _name;
            }      
          
            set//隱含傳遞參數value
            {
                _name = value;
            }
        }             

        public virtual void talk()
        {
            Console.WriteLine("Animal Talk");
        }
    }
}